sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2799,6 +2799,26 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("SPARK-24900: speed up sort when the dataset is small") {
+    withTempView("src") {
+      // the optimization works
```

Comment:
It is hard to tell if this optimization actually works from this test. Can we check if `getSampledArray` returns correct data?

Reply:
UT for RangePartitioner has been added.